tests/common.py

```diff
@@ -36,6 +37,33 @@
 DB_MIGRATION_COLLECTION = "migration_info"
 DB_VERSION = "1.2.1"
 
+MONGO_LOCK_FILE = "/mongo-lock"
+
+
+class Lock:
+    def __init__(self):
+        letters = string.ascii_lowercase
+        self.hash = "".join(random.choice(letters) for i in range(128))
+
+    def __enter__(self):
+        while os.path.exists(MONGO_LOCK_FILE):
+            time.sleep(0.1)
+        with open(MONGO_LOCK_FILE, "w") as fh:
+            fh.write(self.hash)
+        return self
+
+    def __exit__(self, type, value, traceback):
+        self.unlock()
+
+    def unlock(self):
+        if not os.path.exists(MONGO_LOCK_FILE):
+            return
+        with open(MONGO_LOCK_FILE, "r") as fh:
+            hash = fh.read()
+            if hash != self.hash:
+                return
+        os.remove(MONGO_LOCK_FILE)
+
```

Comment:
I'm not sure if I get this. The acceptance test has to run serially are you using xdist locally?

Reply:
Class name does indicate what kind of lock this is. Consider renaming it to `MongoLock`.

Reply:
you know, they are using the same db, and I have seen strange things. so I figured there should be some locking.
I thought of giving Lock() a parameter in the constructor, perhaps you like it?